hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/DataSerializableSerializer.java

```diff
@@ -53,6 +54,7 @@
     public static final byte EE_FLAG = 1 << 1;
 
     private static final String FACTORY_ID = "com.hazelcast.DataSerializerHook";
+    private static final Version VERSION = Version.of(BUILD_INFO.getVersion());
```

Comment:
Should we use `BuildInfoProvider.getBuildInfo()` instead? Since version can be overridden via system property.